test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -164,8 +179,8 @@ public void AutoFitRankingTest()
                 RunDetail<RankingMetrics> bestRun = experimentResults[i].BestRun;
                 Assert.True(experimentResults[i].RunDetails.Count() > 0);
                 Assert.NotNull(bestRun.ValidationMetrics);
-                Assert.True(experimentResults[i].RunDetails.Max(i => i.ValidationMetrics.NormalizedDiscountedCumulativeGains.Max() > .4));
-                Assert.True(experimentResults[i].RunDetails.Max(i => i.ValidationMetrics.DiscountedCumulativeGains.Max() > 20));
+                Assert.True(experimentResults[i].RunDetails.Max(i => i.ValidationMetrics.NormalizedDiscountedCumulativeGains.Max() > maxNdcg));
+                Assert.True(experimentResults[i].RunDetails.Max(i => i.ValidationMetrics.DiscountedCumulativeGains.Max() > maxDcg));
```

Comment:
Currently I think you're checking the maximum value of NDCG@N for any `@N`, and for all created models. Instead I think we should check a specific `@N`. 

I didn't review any of the ranking additions to AutoML, is there an input for which `@N` to optimize? If so, I'd check that one.

If we don't know which N the user wants to optimize, I'd check the largest `@N` in NDCG@N:
`bestRun.ValidationMetrics.NormalizedDiscountedCumulativeGains.Last() > 0.6` instead of `Max()`.

This also only checks the one best model (ignoring the others -- removing the fori loop), and checks the NDCG@N for the largest `@N` specified.

I'm assuming the last array element is the NDCG@N and not the first; otherwise it'd be `bestRun.ValidationMetrics.NormalizedDiscountedCumulativeGains[0]`.

**Why check just the one `@N` value?**
We would expect as the `@N` increases, the metric should increase too. I'm not sure it's good to compare against a _single_ value for all `@N` values. This will allow a tighter bound on the expected value.

**Why check just the best model?**
In random sweeping there could easily be a very bad model, for instance if `NumTrees` get set to 2, it may make a poor model.